_posts/2020-12-01-configuring-and-suppressing-warnings.md

```diff
@@ -0,0 +1,192 @@
+* `scalac -W` (`-Y` in 2.12) lists a few additional warnings, such as `-Wdead-code` (`-Ywarn-dead-code` in 2.12).
+
+Finally, warnings can be globally disabled using `-nowarn` or turned into errors with `-Werror` (`-Xfatal-warnings` in 2.12). The new `-Wconf` compiler option allows for  more fine-grained configuration.
+
+## Configuring warnings
+
+The `-Wconf` compiler option allows filtering compiler warnings and applying an action to messages matching the filter. For example, the default configuration
+
+```
+-Wconf:cat=deprecation:ws,cat=feature:ws,cat=optimizer:ws
+```
+
+defines that warnings with category `deprecation` should be summarized as a single warning (`ws`, which means `warning-summary`), and the same for feature and optimizer warnings.
+
+Running `scalac -Wconf:help` explains how to specify a configuration, but we take a detailed look in the following sections. Generally, the syntax is `-Wconf:<filters>:<action>,<filters>:<action>,...`.
+
+### Actions
+
+The `<action>` defines how warnings matching a filter are handled:
+
+* `error` / `e` reports them as errors.
+* `warning` / `w` reports them as warnings (this is the default).
+* `info` / `i` reports them without counting them as warnings, and without causing `-Werror` to fail.
+* `silent` / `s` ignores them.
+
+Like deprecations and feature warnings, a group of `warning`s and `info`s can be reported as a single summary (`warning-summary` / `ws` and `info-summary` / `is`). Specifying `-Wconf:cat=deprecation:w` overrides the default and reports every deprecation warning individually – this is exactly what the `-deprecation` flag does internally.
+
+Warnings and infos can be issued in verbose mode (`warning-verbose` / `wv` and `info-verbose` / `iv`). This displays additional information about the warning that is helpful for writing filters. For example:
+
+```scala
+$> scalac -Wconf:any:wv Test.scala
+Test.scala:4: warning: [deprecation @ my.app.C.f | origin=scala.Predef.ArrowAssoc.→ | version=2.13.0] method → in class ArrowAssoc is deprecated [...]
+  def f() = 1 → 2
+              ^
+```
+
+Here, the warning message includes the following additional information:
+
+* the warning category `deprecation`
+* the site where the warning is issued `my.app.C.f`
+* the origin of the deprecation `scala.Predef.ArrowAssoc.→`
+* the `since` version of the deprecation `2.13.0`
+
+### Filters
+
+The actions explained above are always applied to a set of warnings selected by a filter expression. The following filters are available:
+
+* `any` matches every message.
+* `cat=deprecation` filters according to the message category, for example deprecations (details below). 
+* `msg=regex` applies if some part of the message matches the regex.
+* `site=my\.package\.*` filters on the site where the warning is triggered. The regex must match the entity's full name (`package.Class.method`).
```

Comment:
A note about the backslashes would be welcome. IIUC this is because `.` would otherwise be interpreted as "any character"?

Reply:
There's also an inconsistency between `site` and `origin` below. `site` ends with `\.*`, but `origin` ends with `\..*`. Is that a typo? If not, why is not the same?

Reply:
Thanks, I'll add a note. And it's indeed a typo here.